lib/OpenAI/API/Request.pm

```diff
@@ -27,11 +31,24 @@ has 'user_agent' => (
     builder => '_build_user_agent',
 );
 
+has 'event_loop' => (
+    is      => 'ro',
+    lazy    => 1,
+    builder => '_build_event_loop',
+);
+
 sub _build_user_agent {
     my ($self) = @_;
     $self->{user_agent} = LWP::UserAgent->new( timeout => $self->config->timeout );
 }
 
+sub _build_event_loop {
+    my ($self) = @_;
+    my $class = $self->config->event_loop_class;
+    eval "require $class" or die "Failed to load event loop class $class: $@";
```

Comment:
I usually do this with

```
require "$class.pm" =~ s{::}{/}gr;
```

so as not to have a string-eval in there. Any string-eval is always suspect.

Reply:
That's a good point.

I will probably do it in two steps, because `/r` fails in some old versions of perl.